### PR TITLE
Fix: Image wrong size after upload [ED-1095]

### PR DESCRIPTION
**Pull request by rotemee**
…oad and now loading the correct image dimensions.

## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: includes/widgets/image.php [hunk 1]
```diff
@@ -600,6 +600,37 @@ private function get_caption( $settings ) {
 		return $caption;
 	}
 
+	public function set_svg_size( $image_data, $attachment_id, $size ) {
```
Comment (matipojo) [inline includes/widgets/image.php:603]: What about other widgets with a `media` control?

File: includes/widgets/image.php [hunk 2]
```diff
@@ -600,6 +600,37 @@ private function get_caption( $settings ) {
...
+		$filter = 'wp_get_attachment_image_src';
+
+		if ( 'before_render' === $action ) {
+			if ( $is_custom_size ) {
+				$custom_dimensions = $settings[ 'image_custom_dimension' ];
+
+				add_image_size( 'custom', $custom_dimensions[ 'width' ], $custom_dimensions[ 'height' ] );
+			}
+
+			add_filter( $filter, [ $this, $callback ], 10, 4 );
```
Comment (matipojo) [inline includes/widgets/image.php:624]: Consider use plain strings instead of variables.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by kobizz — CHANGES_REQUESTED** [review]

File: includes/widgets/image.php (hunk 1 above)
Comment (rotemee) [reply includes/widgets/image.php:656]: Fixed - Added support in Image Box, Image Carousel and Image Gallery,

File: includes/widgets/image.php (hunk 2 above)
Comment (rotemee) [reply includes/widgets/image.php:677]: Fixed.

File: includes/widgets/image.php [hunk 3]
```diff
@@ -600,6 +600,37 @@ private function get_caption( $settings ) {
...
+		$image_sizes = Group_Control_Image_Size::get_all_image_sizes();
+
+		$image_data['1'] = $image_sizes[ $size ][ 'width' ];
+		$image_data['2'] = $image_sizes[ $size ][ 'height' ];
+
+		return $image_data;
+	}
+
+	private function handle_svg_image( $action, $settings ) {
+		$is_custom_size= 'custom' === $settings[ 'image_size' ];
```
[the remark this answers is not in the document]
Comment (rotemee) [reply includes/widgets/image.php:666]: Fixed.

File: includes/widgets/image.php (hunk 2 above)
Comment (rotemee) [reply includes/widgets/image.php:677]: Fixed.

File: includes/widgets/image.php [hunk 4]
```diff
@@ -600,6 +600,37 @@ private function get_caption( $settings ) {
...
+		$callback = 'set_svg_size';
+		$filter = 'wp_get_attachment_image_src';
+
+		if ( 'before_render' === $action ) {
+			if ( $is_custom_size ) {
+				$custom_dimensions = $settings[ 'image_custom_dimension' ];
+
+				add_image_size( 'custom', $custom_dimensions[ 'width' ], $custom_dimensions[ 'height' ] );
```
[the remark this answers is not in the document]
Comment (rotemee) [reply includes/widgets/image.php:674]: Fixed - Removed.

File: assets/dev/js/editor/utils/images-manager.js [hunk 5]
```diff
@@ -154,6 +154,30 @@ ImagesManager = function() {
...
 
 	self.debounceGetRemoteItems = _.debounce( self.getRemoteItems, debounceDelay );
+
+	self.isSvgImage = function( imageUrl ) {
+		return 'svg' === imageUrl.split( '.' ).pop();
+	};
+
+	self.getImageSizeFromControlOptions = function( imageSize, sizesOptions, imageCustomDimension ) {
+		var imageData = { width: '', height: '' };
+		var imageDimensions;
```
[the remark this answers is not in the document]
Comment (rotemee) [reply assets/dev/js/editor/utils/images-manager.js:164]: Fixed (to `let`).

File: assets/dev/js/editor/utils/images-manager.js [hunk 6]
```diff
@@ -154,6 +154,30 @@ ImagesManager = function() {
...
+
+		if ( 'custom' === imageSize ) {
+			imageData = imageCustomDimension;
+		} else {
+			imageDimensions = sizesOptions[ imageSize ].match( /(?<=-\s)(\d*\sx\s\d*)/gi );
```
[the remark this answers is not in the document]
Comment (rotemee) [reply assets/dev/js/editor/utils/images-manager.js:169]: Fixed - added to the code in a comment.

**Pull request closed without merging**